subiquity/autoinstall.py

```diff
@@ -0,0 +1,28 @@
+# Copyright 2019 Canonical, Ltd.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import yaml
+
+from curtin.config import merge_config
+
+
+def merge_autoinstall_configs(source_paths, target_path):
```

Comment:
The order of the source_paths controls what the final config will look like due to how merge_config works with dictionaries and lists.  I suggest documenting that behavior.  Additionally it's important to retain the _order_ of the configs, or the order of the expected precedence of autoinstall file locations.  For example, would you expect the initrd-autoinstall.yaml to override other paths?

Reply:
Yes, we need to make some decisions and document how all this works. I'm going to post to discource, probably tomorrow, about this.